src/main/java/com/ocelotslovebirds/birdhaus/blocks/BirdhouseBlockEntity.java

```diff
@@ -69,21 +75,49 @@ public void load(CompoundTag tag) {
      */
 
     public void tickServer() {
-        if (counter > 0) {
-            counter--;
-            setChanged();
+        handleSeedsForTick();
+        handleBirdSpawnForTick();
+    }
+
+    /*
+     * @param xOffset  x offset (East-West axis) at which to spawn the bird
+     * @param yOffset  y offset (Up-Down axis) at which to spawn the bird
+     * @param yOffset  z offset (North-South axis) at which to spawn the bird
+     * All offsets are relative to the birdhouse
+     * TODO: Check if yOffset actually works, it is possible that
+     * birds can only be spawned on the ground.
+    */
+    private void spawnNewBird(int xOffset, int yOffset, int zOffset) {
+        // Dirty type casting because it works
+        ServerLevel lvl = (ServerLevel) this.getLevel();
+        Parrot newBird = new Parrot(EntityType.PARROT, lvl);
+        lvl.addFreshEntity(newBird);
+        BlockPos desBlockPos = this.getBlockPos().offset(xOffset, yOffset, zOffset);
+
+        // Move the spawned bird to the destination
+        // 0.0 because it works, not sure what this supposed to be
+        newBird.moveTo(desBlockPos, (float) 0.0, (float) 0.0);
+    }
+
+    private void handleBirdSpawnForTick() {
+        if (tickerForBirdSpawns.tick()) {
+            int xOffset = ThreadLocalRandom.current().nextInt(-20, 20);
+            int yOffset = 50;
+            int zOffset = ThreadLocalRandom.current().nextInt(-20, 20);
```

Comment:
The y offset doesn't seem to work, I think its possible that birds have to be spawned on the ground (can you guys test this out?)

Reply:
Will test this out tonight

Reply:
You can set the y offset to 
this.getBlockPos().getY()

This will set the Y of the offset to the same Y level the birdhouse is on.

Reply:
@chris7neves hmm good point, I should try `this.getBlockPos().getY() + 20` or something to give the impression that the birds fly down from the sky